### PR TITLE
Remove interop modules

### PR DESCRIPTION
**Pull request by NeQuissimus**
The build for the docs currently fails because the `1.0-RC5` interop modules have not had their package names changed.
The docs assume that they have.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -334,21 +175,21 @@ lazy val docs = project.module
...
     libraryDependencies ++= Seq(
-      "com.github.ghik"     %% "silencer-lib"             % "1.4.1"  % "provided",
-      "commons-io"          % "commons-io"                % "2.6"    % "provided",
-      "org.reactivestreams" % "reactive-streams-examples" % "1.0.2"  % "provided",
-      "org.jsoup"           % "jsoup"                     % "1.12.1" % "provided"
+      "com.github.ghik"     %% "silencer-lib"                       % "1.4.1" % "provided",
+      "commons-io"          % "commons-io"                          % "2.6" % "provided",
+      "org.reactivestreams" % "reactive-streams-examples"           % "1.0.2" % "provided",
+      "org.jsoup"           % "jsoup"                               % "1.12.1" % "provided",
+      "org.scalaz"          %% "scalaz-zio-interop-cats"            % "1.0-RC5",
```
Comment (softinio) [inline build.sbt]: Shouldn't these be from the `zio` org ?

**Comment by iravid** [comment]
Very good. I'm in favor

**Comment by kaishh** [comment]
I'm strongly in favor of monorepos. Splitting will no doubt reduce velocity and make development harder, but whatever :\

**Comment by iravid** [comment]
@NeQuissimus can you detail the motivation for the change?

File: build.sbt (hunk 1 above)
Comment (mijicd) [reply build.sbt]: @softinio Migration in progress :)

**Comment by NeQuissimus** [comment]
The motivation here is multi-faceted:

- Accelerate zio-core adoption of 2.13
- Allow interop modules to issue updates for their non-ZIO dependencies (we don't want a new ZIO version when cats has an update)
- Improve build times

**Comment by kaishh** [comment]
@NeQuissimus 
> Accelerate zio-core adoption of 2.13

zio-core may get to 2.13 faster, but updating the countless interop repositories will be no fun at all, they may eventually start lagging behind in updates.

My biggest concern is with diffusion of momentum, issue tracking and API-breaking changes across separate repositories. If we expect the API to freeze and maintenance effort to move out of core ZIO into the ecosystem, then maybe we're ready for that, otherwise splitting puts more work on the maintainers. I'm just putting it out there because I have a bad feeling about this :o

**Comment by NeQuissimus** [comment]
@jdegoes

**Comment by jdegoes** [comment]
I think we can defer this momentarily, but it's probably going to have to happen very soon.

The main reason is that a change to Cats Effect, Scalaz, or other interop package cannot result in us bumping the ZIO versioning. That means every interop module needs to be versioned independently.

Now, an alternative to splitting out the modules would be to independently version them inside the same (mono) repo. That would require more work around release process.

**Comment by NeQuissimus** [comment]
@mijicd ^^^

**Comment by softinio** [comment]
Must admit I am 100% for moving out the interops. 

Let me know how I can help with this initiative. 

Thanks.

**Comment by NeQuissimus** [comment]
@softinio as I started this pr, @mijicd worked on the new repos. Try syncing up with him. We need the new interops published before the docs here will build.

**Comment by mijicd** [comment]
@NeQuissimus @softinio all of the interops are on their repositories, but, as wrote in Gitter channel, we have different problem with some of them. To be more specific, `interop-cats`, `interop-monix`, `interop-scalaz` and `interop-twitter` depend on `TestRuntime` which is, unfortunately, present only in ZIO's core. There are two ways to resolve this: publish ZIO's test as part of the artifact, or move `TestRuntime` to `testkit`. I would prefer to do the latter, and I will need some help with ZIO release and version upgrades. Reach out on Gitter and we can tackle it together.

**Comment by NeQuissimus** [comment]
@mijicd I think the test artifacts have now been released

**Comment by softinio** [comment]
@NeQuissimus When we did that last release so that the test artifacts are included for some reason `2.11` ones where never propogated to maven. Thats the current blocker. 

As an attempt to overcome it without have to do another release I just tried to do a 2.11 release only via laptop as opposed to CI and got this error:

```
[error] java.lang.RuntimeException: The release task was not run inside the CI.
[error] 	at scala.sys.package$.error(package.scala:26)
[error] 	at ch.epfl.scala.sbt.release.ReleaseEarly$Defaults$.$anonfun$releaseEarly$2(ReleaseEarlyPlugin.scala:308)
[error] 	at sbt.std.Transform$$anon$3.$anonfun$apply$2(System.scala:46)
[error] 	at sbt.std.Transform$$anon$4.work(System.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:269)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error] 	at sbt.Execute.work(Execute.scala:278)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:269)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error] 	at java.lang.Thread.run(Thread.java:745)
[error] (releaseEarly) The release task was not run inside the CI.
[error] Total time: 0 s, completed Jun 14, 2019 7:02:06 PM
```

Unless you know a way around this or a better way to release 2.11 we may have to do another release to proceed. 

Let me know your thoughts.

**Comment by NeQuissimus** [comment]
`export CI=""` does the trick

**Comment by softinio** [comment]
> `export CI=""` does the trick

That did it but now I am getting:

```
[error] Unable to find credentials for [Sonatype Nexus Repository Manager @ oss.sonatype.org].
[warn] CLIENT ERROR: 401 Unauthorized. Check your resolvers username and password.
```

@NeQuissimus Are the sonatype creds you have on our github team page still valid?

**Comment by softinio** [comment]
Credentials look fine to me comparing it whats not hashed in circle settings

**Comment by NeQuissimus** [comment]
The creds should not have changed

**Comment by jdegoes** [comment]
@NeQuissimus What remains to get this one through?

**Comment by mijicd** [comment]
@jdegoes build for 2.11 is failing, thus preventing release of interops.

**Comment by jdegoes** [comment]
@mijicd 

So it looks like we have to:

1. Release 2.11
2. Release interop
3. Update docs to point to released interop
4. Finally, merge in this PR which will delete the interop modules

Sound about right?

**Comment by mijicd** [comment]
@jdegoes Exactly, even the order is right. 

I'd suggest to push the 2.11 asap, as there has been some release-related work in the meantime. Interops require minimal work, basically version bump and docs syncing for cats afaik. I'm not sure how much effort is required to migrate ZIO docs.

**Comment by jdegoes** [comment]
@mijicd Is the ordinary release process working, i.e. if I cut a new RC, will 2.11 get built along with everyone else? Or do we still have problems with 2.11 releasing?

**Comment by mijicd** [comment]
@jdegoes It doesn't seem to be released anymore. At this point, 2.11 is stucked at `1.0.0-RC8-4`. Maybe @NeQuissimus knows more details. As far as I know, he and @softinio are working on fixing the issue.

**Comment by NeQuissimus** [comment]
I know nothing :D

**Comment by ghostdogpr** [comment]
@NeQuissimus any chance you can have a look at https://github.com/zio/zio/issues/1011 ? It seems we need a new SSH key to fix the docs deployment

**Comment by ghostdogpr** [comment]
2.11 is now published without issue. I'm still working on the microsite deployment failing.

**Comment by NeQuissimus** [comment]
This is ready to go :tada:

**Comment by jdegoes** [comment]
@NeQuissimus @mijicd @softinio Thank you all for your help in getting us to this point!

**Pull request merged**